src/graphics/rc_sprite.rs

```diff
@@ -0,0 +1,188 @@
+use crate::{
+    ffi::graphics as ffi,
+    graphics::{
+        Color, Drawable, FloatRect, IntRect, RcTexture, RenderStates, RenderTarget, Texture,
+        Transform, Transformable,
+    },
+    sf_box::SfBox,
+    system::Vector2f,
+};
+use std::{
+    cell::RefCell,
+    ptr::{self, NonNull},
+    rc::Weak,
+};
+
+#[derive(Debug)]
+pub struct RcSprite {
+    sprite: NonNull<ffi::sfSprite>,
+    texture: Weak<RefCell<SfBox<Texture>>>,
+}
+
+impl RcSprite {
+    #[must_use]
+    pub fn new() -> Self {
+        let sp = unsafe { ffi::sfSprite_create() };
+        Self {
+            sprite: NonNull::new(sp).expect("Failed to create Sprite"),
+            texture: Weak::new(),
+        }
+    }
+
+    fn texture_exists(&self) -> bool {
+        self.texture.strong_count() != 0
+    }
+
+    fn set_rc_texture(&mut self, texture: &RcTexture) {
+        self.texture = texture.downgrade()
+    }
+
+    #[must_use]
+    pub fn with_texture(texture: &RcTexture) -> RcSprite {
+        let mut sprite = Self::new();
+        sprite.set_texture(&texture, true);
+        sprite
+    }
+
+    #[must_use]
+    pub fn with_texture_and_rect(texture: &RcTexture, rect: IntRect) -> RcSprite {
+        let mut sprite = Self::with_texture(texture);
+        sprite.set_texture_rect(rect);
+        sprite
+    }
+
+    #[must_use]
+    pub fn set_texture(&mut self, texture: &RcTexture, reset_rect: bool) {
+        self.set_rc_texture(&texture);
+        let raw_texture = unsafe {
+            (*self.texture.upgrade().unwrap().as_ptr())
```

Comment:
`unwrap` here should be safe. It's unwrapping the upgraded `Weak` pointer aka Rc. The only case where this would be a problem is if the Rc's strong_pointer count == 0. The only way for this to happen is if the passed in RcTexture's texture attribute's Rc value points to nothing. That can only occur if the RcTexture object has been dropped as there is no RcTexture function that actually modifies the reference counter.

I am 100% this unwrap is safe, but you can do your own evaluation.